web/modules/custom/joinup_core/joinup_core.post_update.php

```diff
@@ -48,3 +48,38 @@ function joinup_core_post_update_0106400(): void {
   $index->reindex();
   $index->indexItems(-1, 'entity:taxonomy_term');
 }
+
+/**
+ * Update the values of current collections new member role.
+ */
+function joinup_core_post_update_0106401(&$sandbox) {
+  $ids = \Drupal::entityTypeManager()->getStorage('rdf_entity')
+    ->getQuery()
+    ->condition('rid', 'collection')
+    ->execute();
+
+  $graphs = [
+    '<http://joinup.eu/collection/published>',
+    '<http://joinup.eu/collection/draft>',
+  ];
+
+  $connection = \Drupal::getContainer()->get('sparql.endpoint');
+  foreach ($graphs as $graph) {
+    $enabled = <<<QUERY
+WITH {$graph}
+INSERT { ?entity_id <http://joinup.eu/collection/new_role> "rdf_entity-collection-member" }
+WHERE { ?entity_id <http://joinup.eu/collection/moderation> "true"^^<http://www.w3.org/2001/XMLSchema#boolean> }
+QUERY;
+
+    $disabled = <<<QUERY
+WITH {$graph}
+INSERT { ?entity_id <http://joinup.eu/collection/new_role> "rdf_entity-collection-author" }
+WHERE { ?entity_id <http://joinup.eu/collection/moderation> "false"^^<http://www.w3.org/2001/XMLSchema#boolean> }
+QUERY;
+
+    $connection->query($enabled);
+    $connection->query($disabled);
+  }
+
+  \Drupal::entityTypeManager()->getStorage('rdf_entity')->resetCache($ids);
+}
```

Comment:
This is not clear to me, can you explain why this is needed? The non-moderated collections don't need the author role added to function like expected so this update hook seems to be useless.

It is also not asked in the ticket that we should assign the author role to all members of collections that are not moderated.